lib/Core/Inference/MemberTypeResolver.php

```diff
@@ -63,24 +67,26 @@ public function methodType(Type $ownerType, SymbolInformation $info, string $nam
 
     public function constantType(Type $ownerType, SymbolInformation $info, string $name): SymbolInformation
     {
-        $class = $this->reflectClassOrNull($ownerType, $name);
+        $info = $this->attachContainerClass($info, $ownerType, $name);
 
-        if (null === $class) {
+        if (null === $info->containerType()) {
             return $info;
         }
 
         try {
             if (false === $class->constants()->has($name)) {
-                $this->logger->warning(sprintf(
+                $this->logger->warning($message = sprintf(
                     'Class "%s" has no constant named "%s"',
                     (string) $ownerType,
                     $name
                 ));
                 $info = $info->withContainerType(Type::class($class->name()));
+                $info = $info->withError($message);
```

Comment:
The logging is a nice feature - but wonder if it should be removed in most cases in favor of attaching errors to symbol informations..